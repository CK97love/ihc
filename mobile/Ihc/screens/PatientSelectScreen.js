import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  ScrollView,
} from 'react-native';
import {localData} from '../services/DataService';
import PatientTable from '../components/PatientTable';
import Container from '../components/Container';

export default class PatientSelectScreen extends Component<{}> {
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      rows: [],
    };
    this.props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  loadPatients = () => {
    this.setState({ loading: true });
    const data = localData.getPatientSelectRows();
    this.setState({ rows: data, loading: false });
  }

  // Reload table after moving back to table
  onNavigatorEvent(event) {
    if (event.id === 'willAppear') {
      this.loadPatients();
    }
  }

  componentDidMount() {
    this.loadPatients();
  }

  goToPatient = (patient) => {
    this.props.navigator.push({
      screen: 'Ihc.PatientHomeScreen',
      title: patient.name,
      passProps: { name: patient[0], patientKey: patient[7] }
    });
  }

  render() {
    return (
      <Container loading={this.state.loading}>
        <Text style={styles.title}>
          Select a Patient
        </Text>

        <ScrollView contentContainerStyle={styles.tableContainer} horizontal>
          <PatientTable
            rows={this.state.rows}
            goToPatient={this.goToPatient}
          />
        </ScrollView>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  tableContainer: {
    width: '100%'
  },
});
