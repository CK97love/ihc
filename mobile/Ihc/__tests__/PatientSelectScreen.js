import 'react-native';
import React from 'react';
import PatientSelectScreen from '../screens/PatientSelectScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const fakeNavigator = { setOnNavigatorEvent: (param) => {} };
  const json = renderer.create(
    <PatientSelectScreen navigator={fakeNavigator} />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
