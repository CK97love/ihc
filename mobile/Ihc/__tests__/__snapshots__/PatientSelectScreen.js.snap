// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Select a Patient
      </Text>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "width": "100%",
          }
        }
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#dbdbdb",
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                size={2}
                style={
                  Object {
                    "borderWidth": 1,
                    "flex": 2,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Name
                </Text>
              </View>
              <View
                size={1.5}
                style={
                  Object {
                    "borderWidth": 1,
                    "flex": 1.5,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Birthday
                </Text>
              </View>
              <View
                size={1}
                style={
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "column",
                    "maxWidth": 60,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Checkin
                </Text>
              </View>
              <View
                size={1}
                style={
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "column",
                    "maxWidth": 60,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Triage
                </Text>
              </View>
              <View
                size={1}
                style={
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "column",
                    "maxWidth": 60,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Doctor
                </Text>
              </View>
              <View
                size={1}
                style={
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                    "flexDirection": "column",
                    "maxWidth": 60,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Pharmacy
                </Text>
              </View>
              <View
                size={3}
                style={
                  Object {
                    "borderWidth": 1,
                    "flex": 3,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Notes
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 32,
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                onPress={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                    "minHeight": 32,
                  }
                }
              >
                <View
                  size={2}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 2,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    firstname father mother
                  </Text>
                </View>
                <View
                  size={1.5}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1.5,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    01/01/00
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    16:0
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    
                  </Text>
                </View>
                <View
                  size={3}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 3,
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    collapsable={undefined}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#bebebe",
                        "height": "100%",
                        "opacity": 1,
                        "width": "100%",
                      }
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 32,
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                onPress={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                    "minHeight": 32,
                  }
                }
              >
                <View
                  size={2}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 2,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    firstname father mother
                  </Text>
                </View>
                <View
                  size={1.5}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1.5,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    01/01/00
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    16:0
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    
                  </Text>
                </View>
                <View
                  size={1}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 1,
                      "flexDirection": "column",
                      "maxWidth": 60,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  >
                    
                  </Text>
                </View>
                <View
                  size={3}
                  style={
                    Object {
                      "borderWidth": 1,
                      "flex": 3,
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    collapsable={undefined}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#bebebe",
                        "height": "100%",
                        "opacity": 1,
                        "width": "100%",
                      }
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
